### PR TITLE
SAAS-3133: Added MSK Encryption In Transit Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/msk/mskClusterEncryptionInTransit.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                helpers.addResult(results, 0, 'No MSK clusters found', region);
+                return rcb();
+            }
+            
+            for (var cluster of listClusters.data) {
+                if (!cluster.ClusterArn) continue;
+
+                var resource = cluster.ClusterArn;
+
+                if (cluster.EncryptionInfo &&
```
Comment (AkhtarAmir) [inline plugins/aws/msk/mskClusterEncryptionInTransit.js:41]: Let's discuss this tomorrow

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/msk/mskClusterEncryptionInTransit.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'MSK Cluster Encryption In-Transit',
+    category: 'MSK',
+    domain: 'Compute',
```
Comment (sadeed12345) [inline plugins/aws/msk/mskClusterEncryptionInTransit.js:7]:
```suggestion
    domain: 'Compute',
    severity: 'Medium',
```

File: plugins/aws/msk/mskClusterEncryptionInTransit.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    severity: 'Medium',
+    description: 'Ensure that your Amazon MSK clusters have encryption in-transit enabled.',
```
Comment (AkhtarAmir) [inline plugins/aws/msk/mskClusterEncryptionInTransit.js:9]:
```suggestion
    description: 'Ensure that TLS encryption within the cluster feature is enabled for your Amazon MSK clusters.',
```

File: plugins/aws/msk/mskClusterEncryptionInTransit.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    more_info: 'Amazon MSK in-transit encryption is an optional feature, it encrypts data in transit between the brokers of your MSK cluster. You can override this default at the time you create the cluster.',
```
Comment (AkhtarAmir) [inline plugins/aws/msk/mskClusterEncryptionInTransit.js:10]:
```suggestion
    more_info: 'Amazon MSK in-transit encryption is an optional feature which encrypts data in transit within your MSK cluster. You can override this default at the time you create the cluster.',
```

File: plugins/aws/msk/mskClusterEncryptionInTransit.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    link: 'https://docs.aws.amazon.com/msk/latest/developerguide/msk-encryption.html',
+    recommended_action: 'Enable in-transit encryption for MSK clusters',
```
Comment (AkhtarAmir) [inline plugins/aws/msk/mskClusterEncryptionInTransit.js:12]:
```suggestion
    recommended_action: 'Enable TLS encryption within the cluster for all MSK clusters',
```

File: plugins/aws/msk/mskClusterEncryptionInTransit.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            for (var cluster of listClusters.data) {
+                if (!cluster.ClusterArn) continue;
+
+                var resource = cluster.ClusterArn;
+
+                if (cluster.EncryptionInfo && 
+                    cluster.EncryptionInfo.EncryptionInTransit && 
+                    cluster.EncryptionInfo.EncryptionInTransit.InCluster === false) {
+                    helpers.addResult(results, 2,
+                        'Cluster does not have in-transit encryption enabled', region, resource);
```
Comment (AkhtarAmir) [inline plugins/aws/msk/mskClusterEncryptionInTransit.js:46]:
```suggestion
                        'TLS encryption within the cluster is not enabled', region, resource);
```

File: plugins/aws/msk/mskClusterEncryptionInTransit.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                } else {
+                    helpers.addResult(results, 0,
+                        'Cluster has in-transit encryption enabled', region, resource);
```
Comment (AkhtarAmir) [inline plugins/aws/msk/mskClusterEncryptionInTransit.js:49]:
```suggestion
                        'TLS encryption within the cluster is enabled', region, resource);
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**